langchain4j-open-ai/src/main/java/dev/langchain4j/model/openai/OpenAiImageModel.java

```diff
@@ -0,0 +1,149 @@
+     *                       The URL within <code>dev.ai4j.openai4j.image.GenerateImagesResponse</code> will contain
+     *                       the URL to local images then.
+     * @param withPersisting generated response will be persisted under <code>java.io.tmpdir</code>.
+     *                       The URL within <code>dev.ai4j.openai4j.image.GenerateImagesResponse</code> will contain
+     *                       the URL to local images then.
+     */
+    @Builder
+    @SuppressWarnings("rawtypes")
+    public OpenAiImageModel(
+        @NonNull String apiKey,
+        String model,
+        Integer n,
+        String size,
+        String quality,
+        String style,
+        String user,
+        String responseFormat,
+        Duration timeout,
+        Integer maxRetries,
+        Proxy proxy,
+        Boolean logRequests,
+        Boolean logResponses,
+        Boolean withPersisting,
+        Path persistTo
+    ) {
+        timeout = getOrDefault(timeout, ofSeconds(60));
+
+        OpenAiClient.Builder cBuilder = OpenAiClient
+            .builder()
+            .openAiApiKey(apiKey)
+            .callTimeout(timeout)
+            .connectTimeout(timeout)
+            .readTimeout(timeout)
+            .writeTimeout(timeout)
+            .proxy(proxy)
+            .logRequests(getOrDefault(logRequests, false))
+            .logResponses(getOrDefault(logResponses, false))
+            .persistTo(persistTo);
+
+        if (withPersisting != null && withPersisting) {
+            cBuilder.withPersisting();
+        }
+
+        this.client = cBuilder.build();
+
+        this.maxRetries = getOrDefault(maxRetries, 3);
+        this.model = model;
+        this.n = n;
+        this.size = size;
+        this.quality = quality;
+        this.style = style;
+        this.user = user;
+        this.responseFormat = responseFormat;
+    }
+
+    public static class OpenAiImageModelBuilder {
+
+        public OpenAiImageModelBuilder withPersisting() {
+            withPersisting = true;
+            return this;
+        }
+
+        public OpenAiImageModelBuilder logRequests() {
+            logRequests = true;
+            return this;
+        }
+
+        public OpenAiImageModelBuilder logResponses() {
+            logResponses = true;
+            return this;
+        }
+    }
+
+    @Override
+    public Response<List<Image>> generate(String prompt) {
+        GenerateImagesRequest.Builder requestBuilder = GenerateImagesRequest
+            .builder()
+            .prompt(prompt)
+            .size(size)
+            .quality(quality)
+            .style(style)
+            .user(user)
+            .responseFormat(responseFormat);
+
+        if (DALL_E_2.equals(model)) {
+            requestBuilder.model(dev.ai4j.openai4j.image.ImageModel.DALL_E_2);
+        }
+
+        if (n != null) {
+            requestBuilder.n(n);
```

Comment:
minor: this is why I use object types instead of primitives in configs, it is easier to pass them through (no need to check if it is null or not) and also there is an understanding if user defined a config or not (primitive will always have a value).

Reply:
intention here was to use the default value defined in openai4j, so if primitive type is used here, then I have to define a default value here again, right?

Reply:
No, no need to define default value for n, it is an optional parameter. So if you have `Integer n` here, and `Integer n` in openai4j, then you can simply do `.n(n)` and not worry about if it is null or not. If it was not befined by user and is null, then it will have no effect and OpenAI's default value will be used. If user defined it, then it will be used.

Reply:
this part is obsolete now